internal/pkg/runtime/engine/singularity/container_linux.go

```diff
@@ -2094,39 +2166,49 @@ func (c *container) addFuseMount(system *mount.System) error {
 		// opening /dev/fuse before is valid in the RPC server,
 		// where the actual mount operation is going to be
 		// executed.
-		opts := fmt.Sprintf("fd=%d,rootmode=%o,user_id=%d,group_id=%d",
-			cfg.Fuse.DevFuseFd,
+		opts := fmt.Sprintf("fd=%d,rootmode=%o,user_id=%d,group_id=%d%s",
+			fuseMounts[i].Fd,
 			rootmode,
-			os.Getuid(),
-			os.Getgid())
+			uid,
+			gid,
+			allowOther,
+		)
 
 		// mount file system in three steps: first create a
 		// dedicated session directory for each FUSE filesystem
 		// and use that as the mount point.
 		fuseDir := fmt.Sprintf("/fuse/%d", i)
 		if err := c.session.AddDir(fuseDir); err != nil {
-			return err
+			return usernsFd, err
 		}
 		fuseDir, _ = c.session.GetPath(fuseDir)
-		sylog.Debugf("Add FUSE mount %s for %s with options %s", fuseDir, name, opts)
-		if err := system.Points.AddFS(
+
+		sylog.Debugf("Add FUSE mount for %s with options %s", fuseMounts[i].MountPoint, opts)
+		err := system.Points.AddFS(
 			mount.BindsTag,
 			fuseDir,
 			"fuse",
 			syscall.MS_NOSUID|syscall.MS_NODEV,
-			opts); err != nil {
+			opts,
+		)
+		if err != nil {
 			sylog.Debugf("Calling AddFS: %+v\n", err)
-			return err
+			return usernsFd, err
+		}
+
+		if !fuseMounts[i].FromContainer && fakerootHybrid {
+			nsenter := []string{"nsenter", "--user=/dev/fd/4", "-F", "--preserve-credentials"}
```

Comment:
What is `/dev/fd/4` here can you put a comment so it's clear?

stdin, stdout and stderr are 0, 1, and 2... first extra is 3, second extra is 4... but why is it always what we want?